lesson_12/structs_java/structs_app/src/main/java/com/codedifferently/lesson12/Stack.java

```diff
@@ -9,18 +8,30 @@ public Stack() {
   }
 
   public void push(int value) {
-    // Your code here
+    ListNode newNode = new ListNode(value);
+    newNode.next = top;
+    top = newNode;
   }
 
   public int pop() {
-    return 0;
+    int topmostValue = 0;
+    if (isEmpty()) {
+      return Integer.parseInt(null);
```

Comment:
Should throw an exception here since it's impossible to pop an element off of an empty stack.